scipy/_lib/_array_api.py

```diff
@@ -35,29 +35,37 @@ def compliance_scipy(arrays):
     """Raise exceptions on known-bad subclasses.
 
     The following subclasses are not supported and raise and error:
-    - `np.ma.MaskedArray`
+    - `numpy.ma.MaskedArray`
     - `numpy.matrix`
-    - Any array-like which is not Array API compatible or coercible by numpy
-    - object arrays
+    - NumPy arrays which do not have a boolean or numerical dtype
+    - Any array-like which is neither array API compatible nor coercible by NumPy
+    - Any array-like which is coerced by NumPy to an unsupported dtype
     """
     for i in range(len(arrays)):
         array = arrays[i]
         if isinstance(array, np.ma.MaskedArray):
-            raise TypeError("'numpy.ma.MaskedArray' are not supported")
+            raise TypeError("'numpy.ma.MaskedArray' arrays are not supported.")
         elif isinstance(array, np.matrix):
-            raise TypeError("'numpy.matrix' are not supported")
+            raise TypeError("'numpy.matrix' arrays are not supported.")
```

Comment:
this probably should not say `arrays` here, but I don't like how it sounds without it - `matrices` instead?

Reply:
I didn't think so either, but chose not to nit : ) 

"Inputs of type `numpy.matrix` are not supported." or "`numpy.matrix` objects are not supported."